src/tools/parse-dict-header.js

```diff
@@ -0,0 +1,74 @@
+// This function splits a comma-separated string described by RFC 2068 Section 2.
+// Ported from https://github.com/python/cpython/blob/f081fd83/Lib/urllib/request.py#L1399-L1440
+const parseHttpList = s => {
+  const res = [];
+  let part = '';
+
+  let escape = false,
+    quote = false;
+
+  for (let i = 0; i < s.length; i++) {
+    const cur = s.charAt(i);
+    if (escape) {
+      part += cur;
+      escape = false;
+      continue;
+    }
+    if (quote) {
+      if (cur === '\\') {
+        escape = true;
+        continue;
+      } else if (cur === '"') {
+        quote = false;
+      }
+      part += cur;
+      continue;
+    }
+
+    if (cur === ',') {
+      res.push(part);
+      part = '';
+      continue;
+    }
+
+    if (cur === '"') {
+      quote = true;
+    }
+
+    part += cur;
+  }
+
+  if (part) {
+    res.push(part);
+  }
+
+  return res.map(x => x.trim());
+};
+
+// Parse lists of key, value pairs as described by RFC 2068 Section 2 and convert them
+// into a python dict.
```

Comment:
```suggestion
// into an associative array.
```

May be worth swap the "dict" terminology of this file (name of the file and the `parseDictHeader`) into something more common in JS.